### ISSUE TITLE
Ability to extend cache with the cache file

### ISSUE THREAD

**Issue by Strat-** [mpv-player/mpv#2084]
I'd like mpv to continue caching the video I'm streaming to file once the RAM cache has been filled up.

**Comment by yg-yu** [comment]
+1.
and cache-file-size could be set 0 (no limit ) or auto (equal file size);

**Comment by wm4** [comment]
>and cache-file-size could be set 0 (no limit ) or auto (equal file size);

This makes not much sense. --cache-file-size merely sets the maximum, and the cache file is extended to this size as needed. The maximum that can be set is about 2TB, is that not enough?

**Comment by Parent5446** [comment]
Maybe I'm a bit confused, but what exactly is the point of the current file cache? If it doesn't have the functionality described in this request, what is it doing other than just storing a second, unused copy of what is already in RAM? In what situation would a file cache be useful unless it is an extension of the in-RAM cache?

**Comment by SEAPUNK** [comment]
@Parent5446 Aside it from being an extension to the in-RAM cache, it's also particularly useful for crappy internet connections.

Think of it as downloading the playing video as you're streaming it: Most of the time, you probably won't have enough RAM to cache entire movies in, so it makes more sense to download the video contents to file, so if your internet connection breaks or slows down for whatever reason, assuming you make your file cache significantly larger than your RAM cache, you have an extra 30 minutes of the video ready to go, rather than the 40-50 seconds that the RAM cache can hold.

A player that I'm modeling this feature against that already does this is KMPlayer, where it essentially downloads the video as it plays it.

**Comment by Parent5446** [comment]
@SEAPUNK So wait, are you saying that's the _current_ functionality? Because if so that makes sense, and it sounds like this bug can be closed. But from reading #2083 and your original feature request, it sounds like that is _not_ the current functionality, but rather the desired goal (which I completely agree with; it makes total sense for the disk cache to be larger than the in-RAM cache).

My comment was saying, if that's not how the cache currently works, i.e., like you described in #2083, then what the hell is the point of the existing file cache? It sounds to me like it currently just duplicates whatever is in the RAM cache and that's it, which makes no sense because what would you use a disk cache for if the entirety of it is in RAM?

**Comment by SEAPUNK** [comment]
No, it's not the desired functionality. It's a feature I want mpv to have.
